src/wasm/wasm-binary.cpp

```diff
@@ -1416,7 +1416,8 @@ Type WasmBinaryBuilder::getType(int initial) {
       // FIXME: for now, force all inputs to be nullable
       return Type(getHeapType(), Nullable);
     case BinaryConsts::EncodedType::i31ref:
-      return Type::i31ref;
+      // FIXME: for now, force all inputs to be nullable
+      return Type(HeapType::BasicHeapType::i31, Nullable);
```

Comment:
I'm not sure this would be any simpler because all the other code needs to be prepared to deal with either `(ref null i31)` or `(ref i31)`. Which version is considered "Basic" because it has the `Type::i31ref` shorthand should really only matter for printing, unless I'm missing something.

Reply:
I think it is slightly simpler in the current approach, since this is more consistent with our other behavior, which is to turn all inputs immediately into nullable types. It's true that it looks a little more awkward here, since it's no longer "basic", but it is still applying that simple rule.